### PR TITLE
Update dependencies, add TypeScript definition

### PR DESCRIPTION
**Pull request by BendingBender**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,9 +24,9 @@ module.exports = (selector, options) => {
 
 	let alreadyFound = false;
 	const promise = new PCancelable((resolve, reject, onCancel) => {
-		let raf;
+		let requestedAnimationFrameId;
```
Comment (bfred-it) [inline index.js]: I'd keep this as was

Comment (BendingBender) [reply index.js]: @sindresorhus Should I revert it?

Comment (bfred-it) [reply index.js]: Yeah

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import PCancelable from 'p-cancelable';
+
+export interface Options {
+	/**
+	 * The element that's expected to contain a match.
+	 *
```
Comment (sindresorhus) [inline index.d.ts]: See https://github.com/sindresorhus/meow/pull/116#discussion_r265038725

File: index.test-d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import {expectType} from 'tsd-check';
+import PCancelable from 'p-cancelable';
+import elementReady from '.';
+
+const p = elementReady('#unicorn');
```
Comment (sindresorhus) [inline index.test-d.ts]:
```suggestion
const promise = elementReady('#unicorn');
```

File: package.json [hunk 4]
```diff
@@ -31,13 +32,14 @@
...
 	"devDependencies": {
-		"ava": "*",
-		"delay": "^2.0.0",
-		"jsdom": "^11.10.0",
-		"xo": "*"
+		"ava": "^1.3.1",
+		"delay": "^4.1.0",
+		"jsdom": "^14.0.0",
+		"tsd-check": "^0.3.0",
+		"xo": "^0.24.0"
```
Comment (bfred-it) [inline package.json]: Unwanted change? @sindresorhus 

```diff
-		"xo": "^0.24.0"
+  		"xo": "*"
```

Comment (sindresorhus) [reply package.json]: No, I started setting versions for everything a while back. Got too many problems with `*`.

**Pull request merged**